spec/defines/database/read_grant_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+valid = {
+  'grant read on new objects from blah to blah': {
+    database_read_only_username: 'puppetdb-read',
+    database_name:               'puppetdb',
+    schema:                      'public',
+  },
+}
+
+invalid = {
+  'no params': {},
+}
+
+describe 'puppetdb::database::read_grant' do
+  let(:facts) { on_supported_os.take(1).first[1] }
```

Comment:
Why did you limit the tests to one OS instead of all?

Reply:
Because that's the way it was :) The change was a simple refactor to remove the `let(:facts)` blocks all over the spec tests.
Many of the classes, which in some cases are just resource wrappers, there is no purpose to testing all platforms.